saml-core/src/main/java/org/keycloak/saml/common/util/StaxParserUtil.java

```diff
@@ -75,24 +78,58 @@ public static void validate(InputStream doc, InputStream sch) throws ParsingExce
     }
 
     /**
-     * Bypass an entire XML element block from startElement to endElement
+     * Bypass an entire XML element block from startElement to endElement.
+     * It is expected that the {@code xmlEventReader} is positioned at (has not yet read)
+     * the start element of the block it should bypass.
      *
      * @param xmlEventReader
      * @param tag Tag of the XML element that we need to bypass
      *
      * @throws org.keycloak.saml.common.exceptions.ParsingException
      */
     public static void bypassElementBlock(XMLEventReader xmlEventReader, String tag) throws ParsingException {
-        while (xmlEventReader.hasNext()) {
-            EndElement endElement = getNextEndElement(xmlEventReader);
-            if (endElement == null)
-                return;
+        XMLEvent xmlEvent = bypassElementBlock(xmlEventReader);
 
-            if (StaxParserUtil.matches(endElement, tag))
-                return;
+        if (! Objects.equals(((EndElement) xmlEvent).getName().getLocalPart(), tag)) {
```

Comment:
bypassElementBlock(xmlEventReader) can return null which will lead to NPE.
I suggest to test it for null and throw parsing exception (something like "no more markup events").